drivers: sensor: add tmp119 support

jeppenodgaard opened the pull request:
Add support for TMP119. Its registers are compatible with TMP117 except for device id register.

Commit:
drivers: sensor: add tmp119 support

Add support for TMP119. Its registers are compatible with TMP117 except for
device id register.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

yperess reviewed (dismissed):
Consider renaming to tmp11x

Commit:
drivers: sensor: rename tmp116 to tmp11x

The tmp116 sensor driver also supports tmp117 and tmp119. Therefore rename
to indicate that is supports a range of tmp devices.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

jeppenodgaard commented:
@yperess I renamed from tmp116 to tmp11x.

jeppenodgaard commented:
Ping @yperess

MaureenHelm reviewed (changes requested):
Code changes are fine, but the different devicetree compatible needs to be mentioned in the migration guide.

Commit:
doc: releases: migration-guide-4.2: rename tmp116 to tmp11x

tmp116 has been renamed to tmp11x because it also supports tmp117 and
tmp119.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

jeppenodgaard commented:
> Code changes are fine, but the different devicetree compatible needs to be mentioned in the migration guide.

Done.

jeppenodgaard commented on dts/bindings/mtd/ti,tmp11x-eeprom.yaml:
```diff
@@ -1,10 +1,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
-description: TI TMP116 EEPROM controller binding
+description: TI TMP11X EEPROM controller binding
```
Updated as suggested
